.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,98 @@
+
+name: "Continuous Integration"
+
+on:
+  pull_request:
+
+jobs:
+  roave_bc_check:
+    name: "Roave BC Check"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: fetch tags
+        run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+      - name: Roave BC Check
+        uses: docker://nyholm/roave-bc-check-ga
+
+  phpunit:
+    name: "PHPUnit"
+    runs-on: "ubuntu-20.04"
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version:
+          - "7.1"
+          - "7.2"
+          - "7.3"
+          - "7.4"
+          - "8.0"
+        deps:
+          - "normal"
+        include:
+          - deps: "low"
+            php-version: "7.1"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+        with:
+          fetch-depth: 2
+
+      - name: "Install PHP with XDebug"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          php-version: "${{ matrix.php-version }}"
+          coverage: "xdebug"
+          extensions: "pdo_sqlite"
+          ini-values: "zend.assertions=1"
+
+      - name: "Download box"
+        run: "./download-box.sh"
+
+      - name: "Cache dependencies installed with composer"
+        uses: "actions/cache@v2"
+        with:
+          path: "~/.composer/cache"
+          key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}"
+          restore-keys: "php-${{ matrix.php-version }}-composer-locked-"
+
+      - name: "Install dependencies with composer"
+        run: "composer update --no-interaction --prefer-dist"
+        if: "${{ matrix.deps == 'normal' }}"
+
+      - name: "Install lowest possible dependencies with composer"
+        run: "composer update --no-interaction --prefer-dist --prefer-lowest"
+        if: "${{ matrix.deps == 'low' }}"
+
+      - name: "Run PHPUnit"
+        run: "vendor/bin/phpunit --coverage-clover=coverage.xml"
+
+      - name: "Upload coverage file"
```

Comment:
I would recommend uploading the coverage file only once and enable the coverage run only on a specific php version. It will saves build time. Here an example how it could be done https://github.com/humbug/php-scoper/pull/422

Reply:
What if I have code that only execute on a specific version of PHP?

Reply:
is Codecov able to distinguish them? (i remember that scrutinizer was considering only the first received file...)

Reply:
Yes, it can tell you how many jobs cover a single line

Reply:
Nevermind, now unterstand what the coverage steps are doing. Looks fine!

Reply:
It's a bit more complex that uploading directly to Codecov, I had to do it on the DBAL where those upload would fail because of a limit in the number of jobs, and ending up doing this: one upload of all artifacts in one step. It's a bit more complicated but I think it might make the build more stable by reducing the possibility of a failure when uploading to Codecov.